### PR TITLE
Allow to update the app id/secret

### PR DESCRIPTION
**Pull request by kxgen**
usage:
$fb = App::make('SammyK\LaravelFacebookSdk\LaravelFacebookSdk');
$fb->updateAppIdSecret('your_app_id', 'your_app_secret');

### CODE REVIEW

**Comment by SammyK** [comment]
Thanks for the PR!

I think this would be better if we could send in an all new $config array so that different apps could use different options (like different versions of Graph for example). You can just `array_merge()` the existing config so that it will fallback to the original instance config for unspecified options.

Also the method should be named `newInstance()` and return a new instance of itself:

```php
return new static($config);
```

**Comment by SammyK** [comment]
Hey @kxgen! Just wanted to check in with you to see if you'd be able to update your PR based on my feedback. If not, l can get in there and make the patch. :)

**Comment by kxgen** [comment]
Hi @SammyK , sorry to reply you so late, I just follow your guide and change the update method.
So now we can use something like this to update app info:
```
$new_config = ['app_id' => 'you_app_id', 'app_secret' => 'you_app_secret'];
$fb->newInstance($new_config)->post('things you want to do');
```
right?

File: src/LaravelFacebookSdk/LaravelFacebookSdk.php [hunk 1]
```diff
@@ -24,12 +24,23 @@ class LaravelFacebookSdk extends Facebook
...
     }
 
     /**
+     * @param array   $config
+     */
+    public function newInstance(array $config)
+    {
+        $new_config = array_merge($this->default_config, $config);
+
+        return new static($this->config_handler, $this->url, $config);
```
Comment (SammyK) [inline src/LaravelFacebookSdk/LaravelFacebookSdk.php]: You should be passing in `$new_config` as the 3rd arg

File: src/LaravelFacebookSdk/LaravelFacebookSdk.php [hunk 2]
```diff
@@ -24,12 +24,23 @@ class LaravelFacebookSdk extends Facebook
     public function __construct(Config $config_handler, Url $url, array $config)
     {
         $this->config_handler = $config_handler;
-        $this->url = $url;
+        $this->url            = $url;
```
Comment (SammyK) [inline src/LaravelFacebookSdk/LaravelFacebookSdk.php]: No extra spaces between the variable and the `=` here to align it. It makes diffs harder to read.

**Comment by kxgen** [comment]
oops!

File: src/LaravelFacebookSdk/LaravelFacebookSdk.php [hunk 3]
```diff
@@ -39,7 +50,7 @@ public function __construct(Config $config_handler, Url $url, array $config)
      */
     public function getLoginUrl(array $scope = [], $callback_url = '')
     {
-        $scope = $this->getScope($scope);
+        $scope        = $this->getScope($scope);
```
Comment (SammyK) [inline src/LaravelFacebookSdk/LaravelFacebookSdk.php]: No extra spaces here.

File: src/LaravelFacebookSdk/LaravelFacebookSdk.php [hunk 4]
```diff
@@ -55,7 +66,7 @@ public function getLoginUrl(array $scope = [], $callback_url = '')
      */
     public function getReRequestUrl(array $scope, $callback_url = '')
     {
-        $scope = $this->getScope($scope);
+        $scope        = $this->getScope($scope);
```
Comment (SammyK) [inline src/LaravelFacebookSdk/LaravelFacebookSdk.php]: Same here

File: src/LaravelFacebookSdk/LaravelFacebookSdk.php (hunk 2 above)
Comment (kxgen) [inline src/LaravelFacebookSdk/LaravelFacebookSdk.php]: oops

File: src/LaravelFacebookSdk/LaravelFacebookSdk.php [hunk 5]
```diff
@@ -71,7 +82,7 @@ public function getReRequestUrl(array $scope, $callback_url = '')
      */
     public function getReAuthenticationUrl(array $scope = [], $callback_url = '')
     {
-        $scope = $this->getScope($scope);
+        $scope        = $this->getScope($scope);
```
Comment (SammyK) [inline src/LaravelFacebookSdk/LaravelFacebookSdk.php]: And here. :)

**Comment by SammyK** [comment]
Thanks @kxgen! Just change based on my comments and it should be good to pull in. :)

**Comment by kxgen** [comment]
:)

**Comment by SammyK** [comment]
Thanks! :)

**Pull request merged**